controllers/cluster_controller.go

```diff
@@ -286,7 +286,7 @@ func (r *ClusterReconciler) reconcile(ctx context.Context, cluster *apiv1.Cluste
 		return ctrl.Result{}, err
 	}
 
-	if instancesStatus.AllReadyInstancesStatusUnreachable() {
+	if instancesStatus.AllReadyAndActiveInstancesAreUnreachable() {
```

Comment:
IMO, this is one of the root causes of code being extremely hard to understand and could lead to dangerous bugs.

In this check, we are only evaluating Ready and Active instances, but after that we operate on the original list containing the inactive or not ready instances.

We should start thinking of refactoring this in a filter that returns a new object and adapt the following code to it

Reply:
@mnencia @gbartolini @leonardoce